### ISSUE TITLE
Error: Cannot find module '<path>/routes-manifest.json

### ISSUE THREAD

**Issue by SpencerKaiser** [zeit/next.js#11260]
# Bug report
Running the app in a production environment fails because `routes-manifest.json` cannot be found

## Describe the bug
I have a custom Node.js server written in Typescript and I'm trying to deploy to IBM Cloud. Whenever I deploy my `.next` and `dist` directories, I end up getting the following error: `Error: Could not find a valid build in the '<path>/next' directory! Try building your app with 'next build' before starting the server.`

I tried to reproduce this error locally by mimicking the process of building my app in Jenkins and deploying via Cloud Foundry's `cf` cli and I get the following error whenever I try to start the app with `NODE_ENV=production node dist/index.js`: 
```
Error: Cannot find module '<path>/next/routes-manifest.json'
Require stack:
- <path>/node_modules/next/dist/next-server/server/next-server.js
- <path>/node_modules/next/dist/server/next.js
- <path>/dist/app.js
- <path>/dist/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:794:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:849:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Server.getCustomRoutes (<path>/node_modules/next/dist/next-server/server/next-server.js:154:16)
    at Server.generateRoutes (<path>/node_modules/next/dist/next-server/server/next-server.js:167:34)
    at new Server (<path>/node_modules/next/dist/next-server/server/next-server.js:83:49)
    at Object.createServer [as default] (<path>/node_modules/next/dist/server/next.js:2:133)
    at Object.<anonymous> (<path>/dist/app.js:21:31)
    at Module._compile (internal/modules/cjs/loader.js:956:30) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '<path>/node_modules/next/dist/next-server/server/next-server.js',
    '<path>/node_modules/next/dist/server/next.js',
    '<path>/dist/app.js',
    '<path>/dist/index.js'
  ]
```

## To Reproduce

1. Delete `next` and `dist` directories
1. Delete `node_modules`
1. Run `npm ci`
1. Run `npm build` which uses these scripts:
  ```
    "build": "run-s build:ts build:next",
    "build:ts": "tsc -p src/ --pretty",
    "build:next": "next build",
  ```
5. Run `NODE_ENV=production node dist/index.js`

## Expected behavior
The app should start using both my compiled server and the compiled next directories.

## System information

- OS: macOS
- Version of Next.js: 9.3.0

## Additional Information
I've validated that on the IBM Cloud VM both `next` and `dist` directories exist and running `npm run dev` locally starts the app as expected using the following scripts:
```
    "dev": "npm-run-all -l build:ts -p build:watch start:watch",
    "start": "node dist/index.js",
    "start:watch": "nodemon -w dist -e js dist/index.js",
    "build": "run-s build:ts build:next",
    "build:ts": "tsc -p src/ --pretty",
    "build:next": "next build",
    "build:watch": "npm run build:ts -- -w",
```

**Comment by Timer** [comment]
This is due to a mismatch in Next.js versions between build and deploy machine. Be sure your Next.js version is pinned.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
